lib/rubycritic/rake_task.rb

```diff
@@ -0,0 +1,75 @@
+require 'rake'
+require 'rake/tasklib'
+require 'English'
+
+module Rubycritic
+  #
+  # A rake task that runs RubyCritic on a set of source files.
+  #
+  # This will create a task that can be run with:
+  #
+  #   rake rubycritic
+  #
+  # Example:
+  #
+  #   require 'rubycritic/rake_task'
+  #
+  #   RubyCritic::RakeTask.new do |task|
+  #     task.paths = FileList['lib/**/*.rb', 'spec/**/*.rb']
+  #   end
+  #
+  class RakeTask < ::Rake::TaskLib
+    # Name of RubyCritic task. Defaults to :rubycritic.
+    attr_writer :name
+
+    # Glob pattern to match source files. Defaults to FileList['lib/**/*.rb'].
+    attr_writer :paths
+
+    # Use verbose output. If this is set to true, the task will print
+    # the rubycritic command to stdout. Defaults to false.
+    attr_writer :verbose
+
+    # You can pass all the options here in that are shown by "rubycritic -h" except for
+    # "-p / --path" since that is set separately. Defaults to ''.
+    attr_writer :options
+
+    def initialize(name = :rubycritic)
+      @name    = name
+      @paths   = FileList['lib/**/*.rb']
+      @options = ''
+      @verbose = false
+
+      yield self if block_given?
+      define_task
+    end
+
+    private
+
+    attr_reader :name, :paths, :verbose, :options
+
+    def define_task
+      desc 'Run RubyCritic'
+      task(name) { run_task }
+    end
+
+    def run_task
+      puts "\n\n!!! Running #{name} rake command: #{command}\n\n" if verbose
+      system(*command)
```

Comment:
Can't we dog food our own Ruby API instead of going the `system` route? We could start with something like the following (we would still need to support the `path`, `minimum-score` and other options):

```ruby
command = Rubycritic::Command::Default.new(paths: paths)
reporter = command.execute
print(reporter.status_message)
exit reporter.status
```

Or go straight down the `Rubycritic::Cli::Application.new(*(options_as_arguments + paths))` way.